EclEditor/LexSALT.cxx

```diff
@@ -0,0 +1,415 @@
+            }
+            break;
+        case SCE_SALT_FIELDNAME:
+            if (sc.Match(':')) {
+                sc.SetState(SCE_SALT_OPERATOR);
+                sc.ForwardSetState(SCE_SALT_DEFAULT);
+                collectingFieldName = false;
+            }
+            else if (sc.Match(')')) {
+                sc.SetState(SCE_SALT_OPERATOR);
+                collectingFieldName = false;
+                parens--;
+                if (parens == 0) {
+                    sc.SetState(SCE_SALT_DEFAULT);
+                }
+                else {
+                    sc.SetState(SCE_SALT_ARGUMENTS);
+                }
+            }
+            break;
+            /*
+            case SCE_SALT_NUMBER:
+            // We accept almost anything because of hex. and number suffixes
+            if (!setWord.Contains(sc.ch)) {
+            sc.ForwardSetState(SCE_SALT_DEFAULT);
+            }
+            break;
+            */
+        case SCE_SALT_COMMENT:
+            if (sc.Match('*', '/')) {
+                sc.Forward();
+                sc.ForwardSetState(SCE_SALT_DEFAULT);
+            }
+            break;
+        case SCE_SALT_COMMENTDOC:
+            if (sc.Match('*', '/')) {
+                sc.Forward();
+                sc.ForwardSetState(SCE_SALT_DEFAULT);
+            }
+            else if (sc.ch == '@' || sc.ch == '\\') { // JavaDoc and Doxygen support
+                                                      // Verify that we have the conditions to mark a comment-doc-keyword
+                if ((IsASpace(sc.chPrev) || sc.chPrev == '*') && (!IsASpace(sc.chNext))) {
+                    styleBeforeDCKeyword = SCE_SALT_COMMENTDOC;
+                }
+            }
+            break;
+        case SCE_SALT_COMMENTLINE:
+            if (sc.atLineStart) {
+                sc.SetState(SCE_SALT_DEFAULT);
+            }
+            break;
+        }
+
+        // Determine if a new state should be entered.
+        int lineCurrent = styler.GetLine(sc.currentPos);
+        int lineState = styler.GetLineState(lineCurrent);
+        if (sc.atLineStart) {
+            sc.SetState(SCE_SALT_DEFAULT);
+            line++;
+        }
+
+        if (sc.atLineStart) {
+            command = false;
+            fieldName = false;
+            concept = false;
+            collectingFieldName = false;
+        }
+
+        if (sc.state == SCE_SALT_DEFAULT) {
+            if (lineState) {
+                sc.SetState(lineState);
+            }
+            else if (sc.Match(':')) {
+                if (concept && collectingFieldName) {
+                    if (keywords2.InList(s)) {
+                        sc.ChangeState(SCE_SALT_COMMAND);
+                        concept = false;
+                        collectingFieldName = false;
+                    }
+                    else {
+                        sc.ChangeState(SCE_SALT_FIELDNAME);
+                    }
+                }
+                else if (strcmp(s, "concept") == 0){
```

Comment:
It is unique in that it is the only one to take field names separated by colons. It is important to color field names in red.